src/main/java/com/artipie/asto/s3/S3Storage.java

```diff
@@ -215,15 +193,21 @@ public S3Storage(final S3AsyncClient client, final String bucket, final boolean
 
     @Override
     public CompletableFuture<Long> size(final Key key) {
-        return handleNoSuchKey(
-            key,
-            this.client.headObject(
-                HeadObjectRequest.builder()
-                    .bucket(this.bucket)
-                    .key(key.string())
-                    .build()
-            ).thenApply(HeadObjectResponse::contentLength)
-        );
+        final CompletableFuture<Long> future = this.client.headObject(
+            HeadObjectRequest.builder()
+                .bucket(this.bucket)
+                .key(key.string())
+                .build()
+        ).thenApply(HeadObjectResponse::contentLength);
+        return future
```

Comment:
@andreoss there are not need to introduce a one-time variable, you can just call `handle` after `thenApply`

Reply:
@g4s8 Done